### DISCUSSION TITLE
QMP - Dudas de como se generan los atuendos - hasta donde llegar

### DISCUSSION THREAD

**Discussion by francorojo** [dds-jv/foro#109] [Consultas de objetos] [answered]
Hola buenas noches, tengo una duda que me quede pensando sobre como interpretar el dominio y evaluar sus limites. Perdon que se me hace largo explicarlo, seria mas facil en un audio creo.
En el caso de QueMePongo dijimos que crear la clase Usuario o QueMePongo o Negocio no deberian estar, para concentrarnos en los datos que nos interesan realmente guardar. Como las prendas, y suponer que el metodo agregarPrendas() no hace falta porque se realiza por la interface. Como por ejemplo no validar el color, ya que este es apretado por pantalla por el ususario y nos llega su codigo hexadecimal directo. Pero el hecho de generarAtuendos(), considero que tendriamos que realizarlo porque requiere una logica atras y tenemos que resolverlo. Entonces ese metodo quien lo haria? No es un metodo que lo pueda resolver la clase prenda ni sus "derivadas". No se necesita una clase que tenga la lista de prendas de cada usuario? o en su defecto la clase usuario con sus prendas?
Tal vez si concideramos que es una app, habria clase Usuario(habria una instancia para cada usuario nuevo) y no la clase QueMePongo y tal vez si concideramos que es un servicio para un local podria ser la clase QueMePongo (habria una instancia para cada local) que se encargue de todo y no hace falta la clase usuario.
Me quedo mucha duda de estas opciones diferentes que pueden aparecer depende como uno lo interprete.

**Comment by gastonprieto** [comment, marked as answer]
> Pero el hecho de generarAtuendos(), considero que tendriamos que realizarlo porque requiere una logica atras y tenemos que resolverlo. 

Correcto, pero el tema es que en esta iteración aún no tenes que hacer eso, es el objetivo final del sistema, generar atuendos, pero aún no llegaste a resolver esa parte. Por eso hablamos de YAGNI, primero validar las cosas de esta iteración y luego agregaremos lo otro.

> Entonces ese metodo quien lo haria? No es un metodo que lo pueda resolver la clase prenda ni sus "derivadas". No se necesita una clase que tenga la lista de prendas de cada usuario? o en su defecto la clase usuario con sus prendas?

Si, seguramente la tengas, probablemente sea el usuario, o tal vez aparece un requerimiento nuevo que te indica que eso es compartido entonces lo tendría otro objeto, por eso es importante aún no atacar ese problema, porque nos falta información.

Y como bien vos planteas aún podrían ser muchos modelos entonces, hay que bajar la ansiedad y esperar a que el usuario nos complete la información para realizarlo, vamos a seguir trabajando con quemepongo así que seguramente entiendas un poco mas del modelo mas adelante.

**Comment by gastonprieto** [comment]
@francorojo le cambie el titulo mas que nada por si alguien tiene dudas en el futuro tal vez llegue mas rapido

**Comment by francorojo** [comment]
Aaa perfecto, habia entendido que tal vez no se debia hacer incluso mas adelante, ahora tiene sentido. Muchas Gracias!
